src/BloomExe/web/ServerBase.cs

```diff
@@ -1,99 +1,57 @@
-﻿// Copyright (c) 2014 SIL International
+﻿// Copyright (c) 2015 SIL International
 // This software is licensed under the MIT License (http://opensource.org/licenses/MIT)
 using System;
+using System.Collections.Generic;
+using System.Diagnostics;
 using System.IO;
+using System.Linq;
 using System.Net;
 using System.Threading;
 using System.Windows.Forms;
-using Palaso.Reporting;
 using Palaso.Code;
+using Palaso.Reporting;
+using ThreadState = System.Threading.ThreadState;
+
 
 namespace Bloom.web
 {
-	public abstract class ServerBase: IDisposable
+	public abstract class ServerBase : IDisposable
 	{
-		public static bool IsAbleToUsePort;
-
-		private HttpListener _listener;
+		private readonly HttpListener _listener;
+		private readonly Thread _listenerThread;
+		private readonly Thread[] _workers;
+		private readonly ManualResetEvent _stop, _ready;
+		private readonly Queue<HttpListenerContext> _queue;
 		private bool _isDisposing;
-		private Thread _listenerThread;
-		private readonly ManualResetEvent _stop;
 
-		public ServerBase()
+		protected ServerBase()
 		{
+			// limit the number of worker threads to the number of processor cores
```

Comment:
Further testing was inconclusive. It appears that Bloom uses the same amount of memory if I use 4 threads (number of cores) or just 2 threads.